tests/profile.test.js

```diff
@@ -0,0 +1,277 @@
+      .patch('/api/profiles')
+      .set('Authorization', authToken)
+      .send({
+        firstName,
+        lastName,
+        address,
+        bio,
+        gender,
+        profileImage,
+        phoneNumber
+      })
+      .end((err, res) => {
+        res.should.have.status(status.BAD_REQUEST);
+        res.body.errors.should.have.property('country', errors.country);
+        done();
+      });
+  });
+
+  it('should send firstName error', (done) => {
+    chai.request(server)
+      .patch('/api/profiles')
+      .set('Authorization', authToken)
+      .send({
+        country,
+        lastName,
+        bio,
+        address,
+        gender,
+        profileImage,
+        phoneNumber
+      })
+      .end((err, res) => {
+        res.should.have.status(status.BAD_REQUEST);
+        res.body.errors.should.have.property('firstName', errors.firstName);
+        done();
+      });
+  });
+
+  it('should send lastName error', (done) => {
+    chai.request(server)
+      .patch('/api/profiles')
+      .set('Authorization', authToken)
+      .send({
+        country,
+        firstName,
+        bio,
+        address,
+        gender,
+        profileImage,
+        phoneNumber
+      })
+      .end((err, res) => {
+        res.should.have.status(status.BAD_REQUEST);
+        res.body.errors.should.have.property('lastName', errors.lastName);
+        done();
+      });
+  });
+
+  it('should send address error', (done) => {
+    chai.request(server)
+      .patch('/api/profiles')
+      .set('Authorization', authToken)
+      .send({
+        country,
+        firstName,
+        lastName,
+        bio,
+        gender,
+        profileImage,
+        phoneNumber
+      })
+      .end((err, res) => {
+        res.should.have.status(status.BAD_REQUEST);
+        res.body.errors.should.have.property('address', errors.address);
+        done();
+      });
+  });
+
+  it('should send gender error', (done) => {
+    chai.request(server)
+      .patch('/api/profiles')
+      .set('Authorization', authToken)
+      .send({
+        country,
+        firstName,
+        lastName,
+        bio,
+        address,
+        profileImage,
+        phoneNumber
+      })
+      .end((err, res) => {
+        res.should.have.status(status.BAD_REQUEST);
+        res.body.errors.should.have.property('gender', errors.gender);
+        done();
+      });
+  });
+
+  it('should send phoneNumber error', (done) => {
+    chai.request(server)
+      .patch('/api/profiles')
+      .set('Authorization', authToken)
+      .send({
+        country,
+        firstName,
+        lastName,
+        bio,
+        address,
+        profileImage,
+        gender
+      })
+      .end((err, res) => {
+        res.should.have.status(status.BAD_REQUEST);
+        res.body.errors.should.have.property('phoneNumber', errors.phoneNumber);
+        done();
+      });
+  });
+
+  it('should send bio error', (done) => {
+    chai.request(server)
+      .patch('/api/profiles')
```

Comment:
@alainburindi let's use `profile` instead of `profiles` in the route. A profile is generally seen as a singular resource.